src/components/OrderHistory/OrderHistoryList.tsx

```diff
@@ -1,35 +1,57 @@
 import { useState, useEffect } from "react";
-import { NavLink } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
 import { navKeys } from "../../routes/routes";
 
 import { fetchOrderHistory } from "../../common/service/common-service";
 import OrderItem from "./OrderItem";
 
-import { Button, PageHeading } from "../../common/styles/componentsStyles";
+import {
+  Button,
+  Container,
+  EmptyListMsg,
+  PageHeading,
+  VFlexBox,
+} from "../../common/styles/componentsStyles";
 import {
   BtnsContainer,
   StyledOrderHistoryList,
 } from "./OrderHistoryListStyles";
 
 import { OrderInterface } from "../../common/types/common.types";
+import Loader from "../../common/styles/loader";
 
 const OrderHistoryList = () => {
   const [orders, setOrders] = useState<OrderInterface[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const navigate = useNavigate();
 
   useEffect(() => {
     const fetchOrders = async () => {
+      setIsLoading(true);
+
       const data = await fetchOrderHistory();
+
+      if (!data) {
+        setIsLoading(false);
+        return;
+      }
+
       setOrders(data);
+      setIsLoading(false);
     };
```

Comment:
async / await stuff should be handled in a try catch block. It seems like if there's an error thrown, we're gonna get an infinite loading loop. Also, it would be a good idea to handle a potential error somehow, like to display something to the user. It's not necessary in this PR, so maybe a bit later we can think of something. After you're done with the whole application, we can try switching to React Query (library) for example or maybe Redux if you prefer to learn some of it. It's much easier to handle errors/ loading states with those libs than by yourself.